command_line/symmetry.py

```diff
@@ -64,87 +95,147 @@
         logger.info(result)
 
         if params.output.json is not None:
             result.as_json(filename=params.output.json)
 
-        self._export_experiments_reflections(experiments, reflections, result)
-
-    def _export_experiments_reflections(self, experiments, reflections, result):
-        from rstbx.symmetry.constraints import parameter_reduction
-
-        reindexed_experiments = copy.deepcopy(experiments)
-        reindexed_reflections = flex.reflection_table()
         # Change of basis operator from input unit cell to best unit cell
         cb_op_inp_best = result.best_solution.subgroup["cb_op_inp_best"]
         # Get the best space group.
         best_subsym = result.best_solution.subgroup["best_subsym"]
         best_space_group = best_subsym.space_group().build_derived_acentric_group()
-        for i, expt in enumerate(reindexed_experiments):
-            # Set the space group to the best symmetry and change basis accordingly.
-            # Setting the basis to one incompatible with the initial space group is
-            # forbidden, so we must first change the space group to P1 to be safe`.
-            expt.crystal.set_space_group(sgtbx.space_group("P 1"))
-            expt.crystal = expt.crystal.change_basis(cb_op_inp_best)
-            expt.crystal.set_space_group(best_space_group)
-
-            S = parameter_reduction.symmetrize_reduce_enlarge(
-                expt.crystal.get_space_group()
-            )
-            S.set_orientation(expt.crystal.get_B())
-            S.symmetrize()
-            expt.crystal.set_B(S.orientation.reciprocal_matrix())
-            reindexed_refl = copy.deepcopy(reflections[i])
-            reindexed_refl["miller_index"] = cb_op_inp_best.apply(
-                reindexed_refl["miller_index"]
+        # Reindex the input data
+        experiments, reflection_tables = _reindex_experiments_reflections(
+            experiments, reflection_tables, best_space_group, cb_op_inp_best
+        )
+
+    elif params.laue_group is not None:
+        if params.change_of_basis_op is not None:
+            cb_op = sgtbx.change_of_basis_op(params.change_of_basis_op)
+        else:
+            cb_op = sgtbx.change_of_basis_op()
+        # Reindex the input data
+        experiments, reflection_tables = _reindex_experiments_reflections(
+            experiments, reflection_tables, params.laue_group.group(), cb_op
+        )
+
+    if params.systematic_absences.check:
+        logger.info("=" * 80)
+        logger.info("")
+        logger.info("Analysing systematic absences")
+        logger.info("")
+
+        if (params.d_min is Auto) and (result is not None):
+            d_min = result.intensities.resolution_range()[1]
+        elif params.d_min is Auto:
+            d_min = resolution_filter_from_reflections_experiments(
+                reflection_tables,
+                experiments,
+                params.min_i_mean_over_sigma_mean,
+                params.min_cc_half,
```

Comment:
Making a note for myself: we should probably only need to do the resolution filtering once at the beginning, rather than once before Laue group determination and once before systematic absences checks.